app/models/patient.rb

```diff
@@ -122,15 +122,50 @@ def latest_transfer
 
   # Patients who are eligible for reminders
   scope :reminder_eligible, lambda {
-    where(purged: false)
-      .where(pause_notifications: false)
+    monitoring_open
+      .joins(:dependents)
       .where('patients.id = patients.responder_id')
-      .where.not('latest_assessment_at >= ?', Time.now.in_time_zone('Eastern Time (US & Canada)').beginning_of_day)
+      .where.not(preferred_contact_method: ['Unknown', 'Opt-out', '', nil])
+      .where(pause_notifications: false)
+      .where(
+        'patients.isolation = ? '\
+        'OR patients.continuous_exposure = ? '\
+        'OR patients.last_date_of_exposure >= ? '\
+        'OR (patients.last_date_of_exposure IS NULL AND patients.created_at >= ?)',
+        true,
+        true,
+        ADMIN_OPTIONS['monitoring_period_days'].days.ago.beginning_of_day,
+        ADMIN_OPTIONS['monitoring_period_days'].days.ago.beginning_of_day
+      )
+      .where(
+        'patients.latest_assessment_at < ? '\
+        'OR patients.latest_assessment_at IS NULL',
+        Time.now.getlocal('-04:00').beginning_of_day
+      )
+      .where(
+        'patients.last_assessment_reminder_sent <= ? '\
+        'OR patients.last_assessment_reminder_sent IS NULL',
+        12.hours.ago
+      )
       .or(
-        where(purged: false)
-          .where(pause_notifications: false)
+        joins(:dependents)
+          .where(purged: false)
+          .where(head_of_household: true)
           .where('patients.id = patients.responder_id')
-          .where(latest_assessment_at: nil)
+          .where('dependents_patients.id != dependents_patients.responder_id')
+          .where.not(preferred_contact_method: ['Unknown', 'Opt-out', '', nil])
+          .where(pause_notifications: false)
```

Comment:
Are these fields typically set on the dependent or the head of household?

Reply:
As of right now, if the dependent has `pause_notifications` as true, it does not affect the HoH. We have recently disabled the option for users to `pause_notifications` for dependents as they aren't actually the ones receiving notifications.

Reply:
Lines 156 and 157 are still only comparing `preferred_contact_method` and `pause_notifications` of the HoH. Comparisons for dependents (not including HoH) in this scope are done through the `dependents_patients` alias. I made the `ignored_dependent_eligibility_fields_test` helper to test the things we may change with a dependent and still consider them eligible (i.e. ignored fields such as paused notifications and contact method).

I'm essentially looking at this scope as:
- everything before the OR is checking if the HoH or individual that is not part of a household is eligible.
- everything in the OR is checking if the HoH has any eligible dependents.

Reply:
Why do the join with dependents in the first `or` at all then?

Reply:
Why do the join with dependents in the first or at all then?

Reply:
Rails complains if the tables specified in the main query and within the OR are not exactly the same, so we have to do the join in the main part of the query even though we don't use the joined table there. We need the joined table to check dependents within the OR